### PR TITLE
fix(npm): use more relaxed package.json version constraint parsing

### PR DESCRIPTION
**Pull request by dsherret**
Includes:

* https://github.com/denoland/deno_semver/pull/23

### CODE REVIEW

File: cli/args/package_json.rs [hunk 1]
```diff
@@ -75,13 +75,13 @@ pub fn get_local_package_json_version_reqs(
       });
     }
     let (name, version_req) = parse_dep_entry_name_and_raw_version(key, value);
-    let result = VersionReq::parse_from_specifier(version_req);
+    let result = VersionReq::parse_from_npm(version_req);
```
Comment (dsherret) [inline cli/args/package_json.rs:78]: Fix is here. I think this was just a mistake initially.

File: tests/specs/npm/npmrc_basic_auth/package.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "npmrc_test",
   "version": "0.0.1",
   "dependencies": {
-    "@denotest/basic": "1.0.0",
+    "@denotest/basic": "^=1.0.0",
```
Comment (dsherret) [inline tests/specs/npm/npmrc_basic_auth/package.json:5]: I piggy-backed on a random test. It causes the more relaxed parsing to be hit.

File: cli/args/package_json.rs [hunk 3]
```diff
@@ -227,6 +227,26 @@ mod test {
     );
   }
```
Comment (dsherret) [inline cli/args/package_json.rs:228]: Oh, it looks like this was previously purposefully done, but we should try to be as npm compatible as possible.

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**